client/data/marketplace/use-wpcom-plugins-query.ts

```diff
@@ -37,31 +37,39 @@ const fetchWPCOMPlugins = ( type: Type, searchTerm?: string, tag?: string ) => {
 	);
 };
 
+interface WPCOMPluginsResponse {
+	results: Array< { [ plugin: string ]: Plugin } >;
+}
+
+export const getWPCOMPluginsQueryParams = (
+	type: Type,
+	searchTerm?: string,
+	tag?: string
+): [ QueryKey, QueryFunction< WPCOMPluginsResponse > ] => {
+	const cacheKey = getCacheKey( type + searchTerm + tag );
+	const fetchFn = () => fetchWPCOMPlugins( type, searchTerm, tag );
+	return [ cacheKey, fetchFn ];
+};
+
 /**
  * Returns marketplace plugins list filtered by searchterm and type.
  *
  * @param {Type} type Optional The query type
  * @param {string} searchTerm Optional The term to search for
  * @param {string} tag Optional The tag to search for
- * @param {{enabled: boolean, staleTime: number, refetchOnMount: boolean}} {} Optional options to pass to the underlying query engine
+ * @param {UseQueryOptions} queryOptions Optional options to pass to the underlying query engine
  * @returns {{ data, error, isLoading: boolean ...}} Returns various parameters piped from `useQuery`
  */
 export const useWPCOMPlugins = (
 	type: Type,
 	searchTerm?: string,
 	tag?: string,
-	{ enabled = true, staleTime = BASE_STALE_TIME, refetchOnMount = true }: UseQueryOptions = {}
+	queryOptions?: UseQueryOptions
 ): UseQueryResult => {
-	return useQuery(
-		getCacheKey( type + searchTerm + tag + `${ enabled ? 'enabled' : 'disabled' }` ),
-		() => fetchWPCOMPlugins( type, searchTerm, tag ),
-		{
-			select: ( data ) => normalizePluginsList( data.results ),
-			enabled: enabled,
-			staleTime: staleTime,
-			refetchOnMount: refetchOnMount,
-		}
-	);
+	return useQuery( ...getWPCOMPluginsQueryParams( type, searchTerm, tag ), {
```

Comment:
Seems like the query and TS improvements could be part of their own PR.

Reply:
Fair enough. I'll add this improvement in a follow-up PR for all query functions.